addons/event/models/event_event.py

```diff
@@ -282,47 +300,84 @@ def _compute_start_sale_date(self):
-
-    @api.onchange('event_type_id')
-    def _onchange_type(self):
-        if self.event_type_id:
-            self.seats_max = self.event_type_id.default_registration_max
-            if self.event_type_id.default_registration_max:
-                self.seats_availability = 'limited'
-
-            if self.event_type_id.auto_confirm:
-                self.auto_confirm = self.event_type_id.auto_confirm
-
-            if self.event_type_id.use_timezone:
-                self.date_tz = self.event_type_id.default_timezone
-
-            self.is_online = self.event_type_id.is_online
+    @api.depends('is_online')
+    def _compute_address_id(self):
+        for event in self:
+            if event.is_online:
+                event.address_id = False
+            elif not event.address_id:
+                event.address_id = self.env.company.partner_id.id
 
-            if self.event_type_id.use_mail_schedule and self.event_type_id.event_type_mail_ids:
-                self.event_mail_ids = [(5, 0, 0)] + [
+    @api.depends('event_type_id')
+    def _compute_date_tz(self):
+        for event in self:
+            if event.event_type_id.use_timezone:
+                event.date_tz = event.event_type_id.default_timezone
+            if not event.date_tz:
+                event.date_tz = self.env.user.tz or 'UTC'
+
+    @api.depends('event_type_id')
+    def _compute_seats_availability(self):
+        """ Make it separate from ``_compute_from_event_type`` because otherwise
+        a value given at create (see create override) would protect all other fields
+        depending on event type id from being computed (see ``_field_computed``) """
+        for event in self:
+            if event.event_type_id.default_registration_max:
+                event.seats_availability = 'limited'
+            if not event.seats_availability:
+                event.seats_availability = 'unlimited'
+
+    @api.depends('event_type_id')
+    def _compute_from_event_type(self):
+        """ Update event configuration from its event type. Depends are set only
+        on event_type_id itself, not its sub fields. Indeed purpose is to emulate
+        an onchange: if event type is changed, update event configuration. Changing
+        event type content itself should not trigger this method.
+
+        Updated by this method
+          * seats_max -> triggers _compute_seats (all seats computation)
+          * auto_confirm
+          * is_online -> triggers _compute_address_id (address_id computation)
+          * event_mail_ids
+          * event_ticket_ids -> triggers _compute_start_sale_date (start_sale_date computation)
+        """
+        for event in self:
+            if not event.event_type_id:
+                if not event.seats_max:
+                    event.seats_max = 0
+                if not event.is_online:
+                    event.is_online = False
+                if not event.event_ticket_ids:
+                    event.event_ticket_ids = False
```

Comment:
Not sure why you set `event_ticket_ids` to False but not `event_mail_ids`

Reply:
Because event_ticket_ids is used as trigger for other compute methods and its value will be necessary somehow. No field depends on event_mail_ids, meaning having its value undefined until after create is not an issue.